app/controllers/concerns/foreman_puppet/api/import_puppetclasses_common_controller.rb

```diff
@@ -74,7 +74,13 @@ def changed_environments
         end
 
         unless @environments.any?
-          render_message(_('No changes to your environments detected'))
+          # check if environment id/name was valid at all
+          if @env_id && (Environment.authorized(:view_environments).find_by(id: @env_id) || \
+             Environment.authorized(:view_environments).find_by(name: @env_id))
+            render_message(_('No changes to your environments detected'))
+          else
+            render_message(_('The environment with id %s cannot be found.') % @env_id)
```

Comment:
Is it possible to inject html here?
I might be overly cautious here.
@kamils-iRonin: WDYT?

Reply:
I'm not sure, we should be safe here 🤔

Reply:
Not sure, either. The easiest fix is to make the message more general such as `The requested environment cannot be found.`